Templating/TemplateGuesser.php

```diff
@@ -30,13 +30,22 @@ class TemplateGuesser
     protected $kernel;
 
     /**
+     * @var string[]
+     */
+    private $controllerPatterns;
+
+    /**
      * Constructor.
      *
-     * @param KernelInterface $kernel A KernelInterface instance
+     * @param KernelInterface $kernel             A KernelInterface instance
+     * @param string[]        $controllerPatterns Regexps extracting the controller name from its FQN.
      */
-    public function __construct(KernelInterface $kernel)
+    public function __construct(KernelInterface $kernel, array $controllerPatterns = array())
     {
+        $controllerPatterns[] = '/Controller\\\(.+)Controller$/';
```

Comment:
It would be harder to only add new patterns (which is IMO the major use case)... For example in ``DunglasActionBundle`` you will have to use this in your config to not break existing applications using your bundle.
```yaml
templating:
    controller_patterns: [ /Controller\\(.+)Controller$/, /Action\\(.+)Action$/ ]
```

Moreover this is the last pattern checked so it can easily be overridden.